### PR TITLE
Add `no-identical-alternatives-in-ternary` rule

### LINKED ISSUE #2153: Rule proposal:  `no-identical-alternate`
### Description

Enfore clean code.

### Fail

```js
a ? b ? c : 1 : 1
```

### Pass

```js
a && b ? c  : 1
```

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by axetroy**
Close #2153

### CODE REVIEW

File: docs/rules/no-identical-alternate.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Disallows nested ternary expressions with repeated alternatives
```
Comment (sindresorhus) [inline docs/rules/no-identical-alternate.md:1]:
```suggestion
# Disallow nested ternary expressions with repeated alternatives
```

**Comment by sindresorhus** [comment]
I think the rule name should include `ternary`. So maybe `no-identical-alternatives-in-ternary`?

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: rules/no-identical-alternate.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const {isSameReference} = require('./utils/index.js');
+
+const MESSAGE_ID = 'no-identical-alternate';
+const messages = {
+	[MESSAGE_ID]: 'Replace nested ternary expressions with a logical expression.',
```
Comment (sindresorhus) [inline rules/no-identical-alternate.js:7]: Should be more specific about what kind of ternary expression it targets.

Comment (axetroy) [reply rules/no-identical-alternate.js:7]: How about `Replace repeated alternatives of ternary expressions with a logical expression.` or `Simplify repeated alternatives of ternary expressions.` ?

**Comment by axetroy** [comment]
> I think the rule name should include `ternary`. So maybe `no-identical-alternatives-in-ternary`?

I'm not very good with rule naming, but I think this is a better name

File: rules/no-identical-alternatives-in-ternary.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const messages = {
+	[MESSAGE_ID]: 'Replace repeated alternatives of ternary expressions with a logical expression.',
+};
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => ({
+	/** @param {import('estree').ConditionalExpression} node */
+	ConditionalExpression(node) {
+		// Check if the alternate is a ConditionalExpression
+		if (node.alternate && node.consequent.type === 'ConditionalExpression') {
```
Comment (fisker) [inline rules/no-identical-alternatives-in-ternary.js:15]: I think we should go deeper, 

```js
a ? (b ? 1 : 2) : (d ? 1 : 3);
```

```js
a ? (b ? 1 : 2) : (d ? 3 : 1)
```

should also be reported. I'm not sure about the equivalent... but there must be a better way to write?

Comment (axetroy) [reply rules/no-identical-alternatives-in-ternary.js:15]: This is too complicated, I don't think it can be done easily.

Comment (axetroy) [reply rules/no-identical-alternatives-in-ternary.js:15]: I think we should ignore this situation, it is almost impossible to happen in the real world.

Or create another rule that names `no-complex ternary` to disallow it.

**Comment by axetroy** [comment]
If it needs to keep the comments, needs a test:

```js
a /** comment */ ? /** comment */ b : /** comment */ c ? 1 : 1
```

File: rules/no-identical-alternatives-in-ternary.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+			const {test: innerTest, consequent: innerConsequent, alternate: innerAlternate} = node.consequent;
+
+			const {sourceCode} = context;
+
+			if (isSameReference(outerAlternate, innerAlternate)) {
+				context.report({
+					node,
+					messageId: MESSAGE_ID,
+					fix(fixer) {
+						const newExpression = `${sourceCode.getText(outerTest)} && ${sourceCode.getText(innerTest)} ? ${sourceCode.getText(innerConsequent)} : ${sourceCode.getText(outerAlternate)}`;
```
Comment (axetroy) [inline rules/no-identical-alternatives-in-ternary.js:26]: TODO: Here will drop the comment

We should change to another replacement implementation

**Comment by axetroy** [comment]
I have reimplemented it. It is supports multiple lines and complex comments retention.

Now it's ready for review @fisker

File: rules/utils/conditional-expression.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+const {getParenthesizedRange} = require('./parentheses.js');
+
+/**
+Find the index of the question mark in a ConditionalExpression node
+
+@param {import('estree').ConditionalExpression} node
+@param {import('eslint').SourceCode} sourceCode
+@returns {number}
+*/
+function findIndexOfQuestionMarkInConditionalExpression(node, sourceCode) {
```
Comment (fisker) [inline rules/utils/conditional-expression.js:12]:
```js
sourceCode.getTokenAfter(node.test, () => ...isQuestionMarkToken)
```

should enough

File: test/no-identical-alternatives-in-ternary.mjs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+const {test} = getTester(import.meta);
+
+test.snapshot({
+	valid: [
+		'a ? b ? c : 1 : 2',
+		'a && b ? c : 1',
+	],
+	invalid: [
+		'a?b?c:1:1',
```
[the remark this answers is not in the document]
Comment (axetroy) [reply test/no-identical-alternatives-in-ternary.mjs:12]: `a ? b ? 1: c :1` fix to `a && b ? 1 : c`

`a ? 1: b ? c : 1` fix to `a || !b ? 1 : c`

`a ? 1 : (b ? 1 : c)` fix to `(a || b) ? 1 : c`

This implementation is too complicated

**Comment by axetroy** [comment]
Close because it is too complex to implement edge cases perfectly.

**Pull request closed without merging**